finetune/base_models/bert/featurizer.py

```diff
@@ -140,4 +145,279 @@ def bert_featurizer(
+    :param encoder: A TextEncoder object.
+    :param config: A config object, containing all parameters for the featurizer.
+    :param train: If this flag is true, dropout and losses are added to the graph.
+    :param reuse: Should reuse be set within this scope.
+    :return: A dict containing;
+        features: The output of the featurizer_final state.
+    """
+    bert_config = BertConfig(
+        vocab_size=encoder.vocab_size,
+        hidden_size=config.n_embed,
+        num_hidden_layers=config.n_layer,
+        num_attention_heads=config.n_heads,
+        intermediate_size=config.bert_intermediate_size,
+        hidden_act=config.act_fn,
+        hidden_dropout_prob=config.resid_p_drop,
+        attention_probs_dropout_prob=config.attn_p_drop,
+        # max_position_embeddings=config.max_length,
+        max_position_embeddings=config.max_chunk_length,
+        type_vocab_size=2,
+        initializer_range=config.weight_stddev,
+        low_memory_mode=config.low_memory_mode,
+        pos_injection=config.context_injection,
+        reading_order_removed=config.reading_order_removed,
+        anneal_reading_order=config.anneal_reading_order,
+        positional_channels=config.context_channels,
+    )
+
+    initial_shape = tf.shape(input=X)
+    initial_shape = tf.compat.v1.Print(
+        initial_shape, ["initial_shape", initial_shape], summarize=-1
+    )
+
+    batch_size = initial_shape[0]
+    sequence_len = initial_shape[1]
+
+    # [batch_size, sequence_length]
+    X = tf.reshape(X, shape=tf.concat(([-1], initial_shape[-1:]), 0))
+    X.set_shape([None, None])
+    X = tf.compat.v1.Print(X, ["X", X, tf.shape(X)], summarize=-1)
+
+    # Add padding to X to later break out a separate chunk dimension
+    """
+    FIXME Remove comments with math. Those are with these assumptions for validating
+    padding / chunking logic
+    * config.max_chunk_length = 64
+    * config.batch_size_scaler = 4
+    * sequence_len = 300
+    """
+    # Need to effectively "scale" chunk dim to account for division by batch_size_scaler so that
+    # num_chunks is divisible by batch_size_scaler. In some cases this requires padding by a lot
+    # of zeros
+    effective_chunk_dim = config.max_chunk_length * config.batch_size_scaler    # 64 * 4 = 256
+    pad_count = effective_chunk_dim - (sequence_len % effective_chunk_dim)  # 256 - (300 % 256) = 212
+    new_seq_len = sequence_len + pad_count  # 300 + 212 = 512
+    num_chunks = new_seq_len / config.max_chunk_length  # 512 / 64 = 8
+    zero_pad = tf.zeros([batch_size, pad_count], dtype=X.dtype)
+    X = tf.concat([X, zero_pad], axis=1)
+    X = tf.compat.v1.Print(X, ["X_pad", X, tf.shape(X)], summarize=-1)
+
+    if config.num_layers_trained not in [config.n_layer, 0]:
+        raise ValueError(
+            "Bert base model does not support num_layers_trained not equal to 0 or n_layer"
+        )
+
+    if config.anneal_reading_order:
```

Comment:
Not sure that this base model makes sense with DocRep - so you can probably remove this if you wanted to generalise this function? @madisonmay ?

Reply:
What's the inherent incompatibility between DocRep and LongDocClassifier?

Reply:
I'm just making the assumption that DocRep in it's current form won't necessarily give good static features at init.